# Business Performance & Revenue Analytics

## Objective
Analyze retail transactions to evaluate revenue growth, customer retention, and revenue concentration.

## Dataset
UCI Online Retail Dataset
Raw records: 541,909

In [9]:
import pandas as pd

df = pd.read_excel("Online Retail.xlsx")

print("Raw Shape:", df.shape)
df.head()

Raw Shape: (541909, 8)


,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,2010-12-01 08:26:00,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,2010-12-01 08:26:00,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom


## Data Quality Audit

Issues identified:
- Missing CustomerID
- Cancelled invoices
- Negative quantities (returns)
- Zero/negative unit prices

In [10]:
# Remove missing CustomerID
df_clean = df.dropna(subset=["CustomerID"]).copy()

# Remove cancelled invoices
df_clean = df_clean[~df_clean["InvoiceNo"].astype(str).str.startswith("C")]

# Remove negative quantities
df_clean = df_clean[df_clean["Quantity"] > 0]

# Remove zero/negative prices
df_clean = df_clean[df_clean["UnitPrice"] > 0]

# Create revenue column
df_clean["Revenue"] = df_clean["Quantity"] * df_clean["UnitPrice"]

# Convert CustomerID
df_clean["CustomerID"] = df_clean["CustomerID"].astype(int)

# Create monthly column
df_clean["YearMonth"] = df_clean["InvoiceDate"].dt.to_period("M")

print("Cleaned Shape:", df_clean.shape)

Cleaned Shape: (397884, 10)


## Cleaning Summary

- Removed incomplete customer records
- Excluded cancellations and returns
- Removed invalid pricing
- Generated revenue metric
- Prepared monthly aggregation column

Cleaned transactions: 397,884

In [8]:
from google.colab import files
files.download("cleaned_retail_transactions.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>